ci: guard against editing already-released migration scripts (#5420)
paradedb/paradedb: RELEASE.md

## Post-Release Steps

> 1. **Verify** that the GitHub Release and GitHub Tag properly created and that all jobs completed.

Author: ```suggestion
1. **Verify** that the GitHub Release and GitHub Tag properly created and that all jobs completed successfully.
```

> 2. **Release** `paradedb/paradedb-enterprise`, `paradedb/charts` and `paradedb/terraform-paradedb-byoc`. More context to come here as we automate more of the release flow.

Author: ```suggestion
2. **Release** `paradedb/paradedb-enterprise` by following the instructions in the repository's RELEASE.md file.
```

## Post-Release Steps

1. **Verify** that the GitHub Release and GitHub Tag properly created and that all jobs completed.

> 2. **Open a post-release PR** against `main` to bump `Cargo.toml` to the next development version (e.g. `0.25.0` or `0.25.0-rc.1`), and `cargo check` to refresh `Cargo.lock`.

Author: ```suggestion
2. **Open a post-release PR** against `main` to bump `Cargo.toml` to the next development version (e.g. `0.24.2` if the previous version was `0.24.1`), and run `cargo check` to refresh `Cargo.lock`.
```

3. **Merge** that PR so `main` reflects ongoing work.

> 4. **Release** `paradedb/paradedb-enterprise`, `paradedb/charts` and `paradedb/terraform-paradedb-byoc`. More context to come here as we automate more of the release flow.

Author: Move this to step 2, and combine the exdisting step 2 and 3 into a single step 3
